src/status_im/contexts/wallet/collectible/events.cljs

```diff
@@ -83,81 +83,183 @@
-         [request-id]            (get-unique-collectible-request-id 1)]
-     {:db (assoc-in db [:wallet :ui :collectibles :pending-requests] 1)
+   (let [current-viewing-account (-> db :wallet :current-viewing-account-address)]
+     {:fx [[:dispatch [:wallet/request-collectibles-for-account {:address current-viewing-account}]]]})))
+
+(rf/reg-event-fx
+ :wallet/collectible-ownership-update-finished-with-error
+ (fn [{:keys [db]} [{:keys [accounts message chainId]}]]
+   (let [address           (first accounts)
+         pending-chain-ids (-> (get-in db [:wallet :ui :collectibles :updating address])
+                               (disj chainId))]
+     {:db (cond-> db
+            (empty? pending-chain-ids)
+            (update-in [:wallet :ui :collectibles :updating] dissoc address)
+
+            (seq pending-chain-ids)
+            (assoc-in [:wallet :ui :collectibles :updating address] pending-chain-ids))
       :fx [[:dispatch
-            [:wallet/request-new-collectibles-for-account
-             {:request-id request-id
-              :account    current-viewing-account
-              :amount     collectibles-request-batch-size}]]]})))
+            [:wallet/log-rpc-error
+             {:event  :wallet/collectible-ownership-update-finished-with-error
+              :params {:address  address
+                       :chain-id chainId}}
+             message]]]})))
+
+(rf/reg-event-fx
+ :wallet/collectible-ownership-update-finished
+ (fn [{:keys [db]} [{:keys [accounts message chainId]}]]
+   (let [address                (first accounts)
+         {:keys [added]}        (transforms/json->clj message)
+         pending-chain-ids      (get-in db [:wallet :ui :collectibles :updating address])
+         added?                 (some? added)
+         added-for-new-address? (and added? (seq pending-chain-ids))
+         updated-chain-ids      (disj pending-chain-ids chainId)]
+     {:db (cond-> db
+            (empty? updated-chain-ids)
+            (update-in [:wallet :ui :collectibles :updating] dissoc address)
+
+            (seq updated-chain-ids)
+            (assoc-in [:wallet :ui :collectibles :updating address] updated-chain-ids))
+      :fx [(when added-for-new-address?
+             [:dispatch
+              [:wallet/request-collectibles-for-account
+               {:address             address
+                :reset-collectibles? true}]])]})))
+
+(rf/reg-event-fx
+ :wallet/collectibles-data-updated
+ (fn [{:keys [db]} [{:keys [message]}]]
+   (let [collectibles-by-address           (->> message
+                                                transforms/json->clj
+                                                data-store/rpc->collectibles
+                                                (group-by #(-> % :ownership first :address)))
+         update-collectibles-in-account-fn (fn [existing-collectibles updated-collectibles]
+                                             (let [idxd-existing (utils.collection/index-by
```

Comment:
Not suree if `idxd` is a clear reduced form for `indexed` :thinking: 

This is a pure function, so we could use complete names an define this function outside this event handler, wdyt?